napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -646,28 +879,36 @@ def handle_action(self, item, pkg_name, action_name, update=False):
                 if hasattr(item, 'latest_version'):
                     pkg_name += f"=={item.latest_version}"
 
-                widget.set_busy(trans._("updating..."), update)
+                widget.set_busy(trans._("updating..."), action_name, update)
                 widget.action_button.setDisabled(True)
             else:
-                widget.set_busy(trans._("installing..."), update)
-
-            method([pkg_name])
+                widget.set_busy(trans._("installing..."), action_name, update)
+            if version:
+                if installer_choice == 'Conda':
+                    installer = 'mamba'
+                else:
+                    installer = 'pip'
+
+                method([pkg_name], versions=[version], installer=installer)
```

Comment:
Would be nice to rename `method` to something more descriptive. `method` means something very specific in python and I don't want this to be ambiguous. Something like `install_method` even would be better I think.